src/modules/vtol_att_control/pusher.cpp

```diff
@@ -0,0 +1,319 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2015 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file pusher.cpp
+ *
+ * @author Simon Wilks		<simon@uaventure.com>
+ * @author Roman Bapst 		<bapstroman@gmail.com>
+ *
+*/
+
+#include "pusher.h"
+#include "vtol_att_control_main.h"
+
+#define ARSP_BLEND_START 8.0f	// airspeed at which we start blending mc/fw controls
+
+Pusher::Pusher(VtolAttitudeControl *attc) :
+	VtolType(attc),
+	_flag_enable_mc_motors(true),
+	_pusher_throttle(0.0f),
+	_mc_att_ctl_weight(1.0f)
+{
+	_vtol_schedule.flight_mode = MC_MODE;
+	_vtol_schedule.transition_start = 0;
+
+	_params_handles_pusher.front_trans_dur = param_find("VT_F_TRANS_DUR");
+	_params_handles_pusher.back_trans_dur = param_find("VT_B_TRANS_DUR");
+	_params_handles_pusher.pusher_mc = param_find("VT_TILT_MC");
```

Comment:
Yes, that's what I meant by "Currently still sharing the tiltrotor param(s) which needs fixing" above ;)